src/libOpenImageIO/imagebufalgo_xform.cpp

```diff
@@ -1144,4 +1155,255 @@ ImageBufAlgo::rotate(const ImageBuf& src, float angle, string_view filtername,
+            const int y_min = clamp((int)floorf(src_y - filterrad_y), ybegin,
+                                    yend);
+            const int y_max = clamp((int)ceilf(src_y + filterrad_y), ybegin,
+                                    yend);
+
+            src_iter.rerange(x_min, x_max + 1, y_min, y_max + 1, 0, 1);
+
+            memset(sample_accum, 0, nchannels * sizeof(Acc_t));
+            float total_weight = 0.0f;
+            for (; !src_iter.done(); ++src_iter) {
+                const float weight = (*filter)(src_iter.x() - src_x + 0.5f,
+                                               src_iter.y() - src_y + 0.5f);
+                total_weight += weight;
+                for (int idx = 0, chan = roi.chbegin; chan < roi.chend;
+                     ++chan, ++idx) {
+                    sample_accum[idx] += src_iter[chan] * weight;
+                }
+            }
+
+            if (total_weight > 0.0f) {
+                for (int idx = 0, chan = roi.chbegin; chan < roi.chend;
+                     ++chan, ++idx) {
+                    out_iter[chan] = sample_accum[idx] / total_weight;
+                }
+            } else {
+                for (int chan = roi.chbegin; chan < roi.chend; ++chan) {
+                    out_iter[chan] = 0;
+                }
+            }
+            ++out_iter;
+        }
+    });  // end of parallel_image
+    return true;
+}
+
+
+
+static bool
+check_st_warp_args(ImageBuf& dst, const ImageBuf& src, const ImageBuf& stbuf,
+                   int chan_s, int chan_t, ROI& roi)
+{
+    // Validate ST buffer
+    if (!stbuf.initialized()) {
+        dst.error("ImageBufAlgo::st_warp : Uninitialized ST buffer");
+        return false;
+    }
+
+    const ImageSpec& stSpec(stbuf.spec());
+    // XXX: Wanted to use `uint32_t` for channel indices, but I don't want to
+    // break from the rest of the API and introduce a bunch of compile warnings.
+    if (chan_s >= stSpec.nchannels) {
+        dst.errorfmt("ImageBufAlgo::st_warp : Out-of-range S channel index: {}",
+                     chan_s);
+        return false;
+    }
+    if (chan_t >= stSpec.nchannels) {
+        dst.errorfmt("ImageBufAlgo::st_warp : Out-of-range T channel index: {}",
+                     chan_t);
+        return false;
+    }
+    // N.B. We currently require floating-point ST channels
+    if (!stSpec.format.is_floating_point()) {
+        dst.error("ImageBufAlgo::st_warp : ST buffer must be holding floating-"
+                  "point data");
+        return false;
+    }
+
+    if (dst.initialized()) {
+        // XXX: Currently requiring the pixel scales of ST and a preallocated
+        // output buffer to match.
+        if (dst.spec().full_width != stSpec.full_width
+            || dst.spec().full_height != stSpec.full_height) {
+            dst.error("ImageBufAlgo::st_warp : Output and ST buffers must have "
+                      "the same full width and height");
+            return false;
+        }
```

Comment:
To be honest, I'm not entirely sure. My instinct is that we can't rely on iterator clamping, since we want to be able to write outside of the display window.

That said, I think this is currently somewhat meaningless at the moment, since I'm unconditionally resetting `dst` below, so this may go away. I'll circle back to this after figuring out how best to handle/skip the buffer allocation below.

Reply:
What I meant is: iterate over dst, which will be initialized to src's size if it doesn't already exist, but there's no reason for stbuf to need to be exactly the same size. If there's a part of dst (or the iteration roi) that's outside the range of stbuf, then you just use the stbuf's iterator's ability to have a wrap mode. You need not consider it an error for stbuf's full_width to be different from dst. I think?